hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/ProtobufStreamingOutput.java

```diff
@@ -91,15 +89,18 @@ private void writeToStream(CellSetModel model, String contentType, OutputStream
 
   private CellSetModel createModelFromResults(Result[] results) {
     CellSetModel cellSetModel = new CellSetModel();
-    for (Result rs : results) {
-      byte[] rowKey = rs.getRow();
-      RowModel rModel = new RowModel(rowKey);
-      List<Cell> kvs = rs.listCells();
-      for (Cell kv : kvs) {
-        rModel.addCell(new CellModel(CellUtil.cloneFamily(kv), CellUtil.cloneQualifier(kv),
-          kv.getTimestamp(), CellUtil.cloneValue(kv)));
+    for (Result r : results) {
+      if (r.isEmpty()) {
+        continue;
       }
-      cellSetModel.addRow(rModel);
+      Cell firstCell = r.rawCells()[0];
```

Comment:
Could we extract a common method for this? Seems the code is exactly the same with the above code?

Reply:
Yes, I will add a utility method somewhere.

Reply:
Pushed the refactor